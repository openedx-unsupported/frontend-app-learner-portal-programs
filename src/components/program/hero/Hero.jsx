import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import { AppContext } from '@edx/frontend-platform/react';

import './styles/Hero.scss';

function Hero(props) {
  const {
    pageContext: {
      pageBranding,
    },
  } = useContext(AppContext);
  const { title } = props;

  return (
    <div className="hero">
      <div className="background d-flex">
        <div
          className="hero-texture d-none d-lg-block"
          style={{
            borderColor: `${pageBranding.banner_border_color}`,
            backgroundImage: `url(${pageBranding.texture_image})`,
          }}
        />
        <div
          className="hero-img"
          style={{
            borderColor: `${pageBranding.banner_border_color}`,
            backgroundImage: `url(${pageBranding.cover_image})`,
          }}
        />
      </div>
      <div className="container py-3">
        <div className="row">
          <div className="col-xs-9 col-lg-6">
            <div className="hero-box bg-white p-4 d-sm-inline-block" style={{ borderColor: `${pageBranding.banner_border_color}` }}>
              <h1 className="hero-heading m-0">{title}</h1>
            </div>
          </div>
        </div>
        <div className="row hero-logo-wrapper">
          <div className="col-12 col-lg-4 offset-lg-8 text-lg-right">
            <div
              className="hero-logo bg-white pl-3 pr-5 d-sm-inline-block text-left"
              style={{
                borderColor: `${pageBranding.banner_border_color}`,
              }}
            >
              <img
                src={pageBranding.organization_logo.url}
                alt={pageBranding.organization_logo.alt}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

Hero.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Hero;
