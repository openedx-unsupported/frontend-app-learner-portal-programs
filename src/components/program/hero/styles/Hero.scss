@import "~@openedx/paragon/scss/core/core.scss";

.hero {
  position: relative;
  overflow: hidden;

  h1 {
    font-family: 'Inter';
    font-weight: 700;
  }

  .container {
    padding-left: 0;

    @include media-breakpoint-up(sm) {
      & {
        padding-left: 2rem;
      }
    }
  }

  .background {
    position: absolute;
    width: 100%;
    height: 100%;

    .hero-texture,
    .hero-img {
      background-color: #ffffff;
      background-size: cover;
      background-repeat: no-repeat;
      background-position: center;
    }

    .hero-texture {
      flex: 1;
      flex-grow: 0;
      flex-basis: 33.33%;
      border-right-width: 20px;
      border-right-style: solid;
    }

    .hero-img {
      flex: 1;
    }
  }

  .hero-box {
    border-left-width: 20px;
    border-left-style: solid;
  }

  .hero-logo-wrapper {
    @include media-breakpoint-between(sm, lg) {
      & {
        margin-top: 20px;
      }
    }

    .hero-logo {
      border-left-width: 20px;
      border-left-style: solid;

      img {
        max-width: 130px;
      }
    }
  }
}
