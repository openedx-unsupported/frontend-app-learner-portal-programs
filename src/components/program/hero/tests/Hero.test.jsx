/* eslint-disable react/jsx-no-constructed-context-values */
import React from 'react';
import { render } from '@testing-library/react';
import { AppContext } from '@edx/frontend-platform/react';

import Hero from '../Hero';

describe('<Hero />', () => {
  it('renders correctly', () => {
    const pageContext = {
      pageBranding: {
        banner_border_color: '#FFFFFF',
        texture_image: 'https://example.com/avengersx-texture-image-440x400.png',
        cover_image: 'https://example.com/avengersx-cover-image-1000x400.png',
        organization_logo: {
          url: 'https://example.com/avengersx-logo-200x101.png',
          alt: 'Avengers logo',
        },
      },
    };

    const { container: tree } = render(
      <AppContext.Provider value={{ pageContext }}>
        <Hero title="Example Title" />
      </AppContext.Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
