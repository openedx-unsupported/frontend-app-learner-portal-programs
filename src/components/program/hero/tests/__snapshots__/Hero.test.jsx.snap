// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> renders correctly 1`] = `
<div>
  <div
    class="hero"
  >
    <div
      class="background d-flex"
    >
      <div
        class="hero-texture d-none d-lg-block"
        style="border-color: #ffffff; background-image: url(https://example.com/avengersx-texture-image-440x400.png);"
      />
      <div
        class="hero-img"
        style="border-color: #ffffff; background-image: url(https://example.com/avengersx-cover-image-1000x400.png);"
      />
    </div>
    <div
      class="container py-3"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-9 col-lg-6"
        >
          <div
            class="hero-box bg-white p-4 d-sm-inline-block"
            style="border-color: #ffffff;"
          >
            <h1
              class="hero-heading m-0"
            >
              Example Title
            </h1>
          </div>
        </div>
      </div>
      <div
        class="row hero-logo-wrapper"
      >
        <div
          class="col-12 col-lg-4 offset-lg-8 text-lg-right"
        >
          <div
            class="hero-logo bg-white pl-3 pr-5 d-sm-inline-block text-left"
            style="border-color: #ffffff;"
          >
            <img
              alt="Avengers logo"
              src="https://example.com/avengersx-logo-200x101.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
